TaskfMRIAnalysis/scripts/makeSubjectTaskSummary.sh

```diff
@@ -0,0 +1,363 @@
+fi
+
+# Determine whether to run Volume, and set strings used for filenaming
+if [ "$VolumeBasedProcessing" = "YES" ] ; then
+	runVolume=true;
+	ExtensionList="${ExtensionList}nii.gz "
+	ScalarExtensionList="${ScalarExtensionList}volume.dscalar.nii "
+	Analyses="${Analyses}StandardVolumeStats "; # space character at end to separate multiple analyses	
+fi
+
+
+# Determine location where SummaryDirectory should be created
+SummaryDirectory="${ResultsFolder}/${SummaryName}${TemporalFilterString}${SmoothingString}${RegString}${ProcSTRING}${LowPassSTRING}${ParcellationString}_subjectSummary.feat"
+
+# Check if summary directory already exists, and remove old summary directory
+if [ -d "$SummaryDirectory" ]; then
+  echo "Removing previous summary directory from ${SummaryDirectory}..."
+  rm -r "$SummaryDirectory"
+fi
+
+# Create new summary directory
+echo "Creating new summary directory at ${SummaryDirectory}..."
+mkdir -v $SummaryDirectory
+
+# Create log file with arguments from TaskfMRIAnalysis.sh call
+touch ${SummaryDirectory}/TaskfMRIAnalysis.log
+
+
+# Loop over analyses requested (runDense or runParcellation) && (runVolume)
+log_Msg "Loop over analyses requested: ${Analyses}"
+analysisCounter=1;
+for Analysis in ${Analyses} ; do
+	log_Msg "Make Summary for Analysis: ${Analysis}"
+	Extension=`echo $ExtensionList | cut -d' ' -f $analysisCounter`;
+	ScalarExtension=`echo $ScalarExtensionList | cut -d' ' -f $analysisCounter`;
+	mkdir -p ${SummaryDirectory}/${Analysis}
+	analysisCounter=$(($analysisCounter+1))
+
+	# Check if Level2 analysis was requested
+	if [ "$LevelTwofMRIName" != "NONE" ]; then
+		echo "Using Level2"
+		# Check if Level2 outputs are present, else throw error
+		LevelTwoFEATDir="${ResultsFolder}/${LevelTwofMRIName}/${LevelTwofsfName}${TemporalFilterString}${SmoothingString}_level2${RegString}${ProcSTRING}${LowPassSTRING}${ParcellationString}.feat"
+		
+		## Check for Contrasts.txt file in feat directory
+		if [ ! -e "${LevelTwoFEATDir}/Contrasts.txt" ]; then
+			# file is missing, write cope number and contrast name to log
+			echo "ERROR: Cannot find Contrasts list at ${LevelTwoFEATDir}/Contrasts.txt. Verify that Level1 and Level2 analyses completed correctly." >> ${SummaryDirectory}/TaskfMRIAnalysis.log
+			log_Err_Abort "ERROR: Cannot find Contrasts list at ${LevelTwoFEATDir}/Contrasts.txt. Verify that Level1 and Level2 analyses completed correctly."
+		else
+			cp -suv ${LevelTwoFEATDir}/Contrasts.txt ${SummaryDirectory}
+		fi
+		
+		copeCounter=1;
+		cat ${LevelTwoFEATDir}/Contrasts.txt | while read Contrast ; do
+			# Check if necessary files exist in cope${i}.feat directory (cope1 should be mean of two runs)
+			if [ -e "${LevelTwoFEATDir}/${Analysis}/cope${copeCounter}.feat/cope1.${Extension}" ]; then
+				echo "${Analysis}: cope${copeCounter} ($Contrast) exists" >> ${SummaryDirectory}/TaskfMRIAnalysis.log
+			fi
+			for File in {mask,cope1,varcope1,tdof_t1} ; do
```

Comment:
This "brace and commas" syntax is actually special bash syntax that just turns into "words with spaces", and isn't necessary:

`for File in mask cope1 varcope1 tdof_t1; do`

Not sure what people think as far as style is concerned.

Reply:
Changed this in next commit.